x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -4,21 +4,22 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import qs from 'querystring';
-import { HttpFetchQuery } from 'src/core/public';
+import { AlertResultList } from '../../../../../common/types';
 import { AppAction } from '../action';
-import { MiddlewareFactory, AlertListData } from '../../types';
+import { MiddlewareFactory, AlertListState } from '../../types';
+import { isOnAlertPage, paginationDataFromUrl } from './selectors';
 
-export const alertMiddlewareFactory: MiddlewareFactory = coreStart => {
-  const qp = qs.parse(window.location.search.slice(1));
-
-  return api => next => async (action: AppAction) => {
-    next(action);
-    if (action.type === 'userNavigatedToPage' && action.payload === 'alertsPage') {
-      const response: AlertListData = await coreStart.http.get('/api/endpoint/alerts', {
-        query: qp as HttpFetchQuery,
-      });
-      api.dispatch({ type: 'serverReturnedAlertsData', payload: response });
-    }
+export const alertMiddlewareFactory: MiddlewareFactory<AlertListState> = coreStart => {
+  return {
+    middleware: api => next => async (action: AppAction) => {
+      next(action);
+      const state = api.getState();
+      if (action.type === 'userChangedUrl' && isOnAlertPage(state)) {
+        const response: AlertResultList = await coreStart.http.post(`/api/endpoint/alerts`, {
```

Comment:
You can always do this later, but we're going to be removing POST support for this API. If you want to try changing it now, it probably will just require changing this to `.get` and then changing the the line below to `query: paginationDataFromUrl(state)),` instead of `body: JSON.stringify(paginationDataFromUrl(state)),` ... I'm fine with waiting until the next PR though.